### DISCUSSION TITLE
BTC Wallet Settings - Unconfirmed TX - automatic CPFP self-spend-TX

### DISCUSSION THREAD

**Discussion by theDavidCoen** [btcpayserver/btcpayserver#4218] [Ideas / Feature Requests]
I'd like to suggest a feature to add to the BTC Wallet Settings -> Payment, if the user selects "Consider the invoice confirmed when the payment transaction …is Unconfirmed".

I'm thinking of a sub-selection or an additional option in which the user can select to do a CPFP TX which spends the incoming UTXO(s) to the same receiving address.
This way, if the user receives a RBF TX that could be later "cancelled" by the payer (double spent, thanks to wallets such as Blue Wallet or Electrum) or if the Bitcoin network is congested and the transaction risk to be dropped by the mempools, the payee can have the transaction included in a block thanks to a subsequential transaction with higher fees.

This would NOT be a solution to the double spend problem but can mitigate the risk of attacks.

Cons:
- The BTC wallet needs to be a HOT one in order to spend funds and so to have this feature
- The user needs to spend fees: maybe he/she can have an option for the multiplier in the Payment settings. For example: 
once the user selects "is Unconfirmed", he/she gets a new CPFP field and something like "select multiplier:" 2X, 4X, 10X. This way the new CPFP transaction will have 2X, 4X or 10X fees than the original transaction accepted with 0 conf.

**Comment by dennisreimann** [comment]
You describe the how it might work and cons, but what are the actual use cases and pros? To me this seems to have a few caveats and also rather complex to build, while the benefits are unclear.

**Reply by theDavidCoen** [reply]
Thanks for checking.
I see this mainly as a prevention against double-spend "attacks" using RBF and as an additional feature for the stores that accept 0 conf TXs.
I would not give high priority to this, of course, but I think it could be helpful since, as per my experience with end-users:
1. there are still many users that accept 0 conf TXs and get scammed
2. there are companies that accept 0 conf TXs
2. There will be more wallets with "cancel transaction" option in the near future
3. I suppose we will see network congestions in the next years
